### DISCUSSION TITLE
Spam of articles of one user

### DISCUSSION THREAD

**Discussion by tiagon97** [realworld-apps/realworld#699] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/48652792/127566576-ca2a3a5f-3f8a-466c-b422-e87f9dfa9a21.png)

Hello, I wanted write some automation tests using Cypress but when I add article I cant even see him in global feed cause user "van" is propably sending tones of POST requests /articles every second, I know its public app but can u do something with that?

**Comment by geromegrignon** [comment]
Thanks for pointing it out.
Sadly the current maintainer team doesn't have access to the deployed API (https://github.com/gothinkster/realworld/issues/642).
I really hope we can fix it as quickly as possible.

**Comment by mvila** [comment]
Probably related: #700

**Comment by geromegrignon** [comment, marked as answer]
A new API has been deployed with limitations to avoid such exposure of NSFW content: https://gothinkster.github.io/realworld/docs/specs/frontend-specs/api#api-limitations

**Comment by cirosantilli** [comment]
I think an email/Oauth signup system + post limit per user is a reasonable bet to help address this kind of issue, started a discussion at: https://github.com/gothinkster/realworld/discussions/783 as I couldn't find one easily.

I manually blacklisted some Cypress stuff from my implementation :cry: https://github.com/cirosantilli/node-express-sequelize-nextjs-realworld-example-app/blob/7def72741eb4123deed9b5e956967d5ca6ba2f6e/config.js#L23

**Reply by geromegrignon** [reply]
Thanks for the Cypress trick; I'll keep it in mind.

Spams are a problem in 2 situations:
- they add NSFW content: fixed by hiding public content lately
- they don't allow users to see their content: fixed by hiding public content lately

There are also several ways to host your API for the second situation (and the one related to this discussion): https://gothinkster.github.io/realworld/docs/specs/frontend-specs/api (I would be happy to see contributions for additional hosting providers).

While experimenting with solutions about NFSW content, I had to reset the database a decent number of times. After each reset, it only took a few minutes to see spam/NSFW content and unrelated ad content displayed on the demo app.

About the Oauth part, I'll answer on the related issue :)
